crates/uv/src/commands/project/sync.rs

```diff
@@ -881,3 +900,270 @@ fn store_credentials_from_target(target: InstallTarget<'_>) {
         }
     }
 }
+
+/// A report of a project's state and updates to make.
+#[derive(Debug, Serialize)]
+struct ProjectReport {
```

Comment:
Can we report the workspace root too? That's probably the more interesting path for IDEs and similar tools.